From 2044acc7ad358feb933765c8228a859d78f7dd52 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Thu, 27 Feb 2025 23:40:11 +0100
Subject: [PATCH] Bader class as optional parameter

---
 pyiron_vasp/vasp/output.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/pyiron_vasp/vasp/output.py b/pyiron_vasp/vasp/output.py
index fbe0507..6084081 100644
--- a/pyiron_vasp/vasp/output.py
+++ b/pyiron_vasp/vasp/output.py
@@ -414,6 +414,7 @@ def parse_vasp_output(
     sorted_indices: list = None,
     read_atoms_funct: callable = read_atoms,
     es_class=ElectronicStructure,
+    bader_class=Bader,
 ) -> dict:
     """
     Parse the VASP output in the working_directory and return it as hierachical dictionary.
@@ -467,7 +468,7 @@ def parse_vasp_output(
     if os.path.isfile(os.path.join(working_directory, "AECCAR0")) and os.path.isfile(
         os.path.join(working_directory, "AECCAR2")
     ):
-        bader = Bader(working_directory=working_directory, structure=structure)
+        bader = bader_class(working_directory=working_directory, structure=structure)
         try:
             charges_orig, volumes_orig = bader.compute_bader_charges()
         except ValueError:
