From 082628047e8c1f9e42459b45c846d1fa700d4f2a Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Thu, 27 Feb 2025 19:41:51 +0100
Subject: [PATCH] Add get_ediel_sol() function (#28)

---
 pyiron_vasp/vasp/parser/outcar.py | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/pyiron_vasp/vasp/parser/outcar.py b/pyiron_vasp/vasp/parser/outcar.py
index 27a2f8ce..79b25e52 100644
--- a/pyiron_vasp/vasp/parser/outcar.py
+++ b/pyiron_vasp/vasp/parser/outcar.py
@@ -484,6 +484,31 @@ def get_energy_sigma_0_from_line(line):
             [get_energy_sigma_0_from_line(lines[j + 4]) for j in trigger_indices]
         )
 
+    @staticmethod
+    def get_ediel_sol(filename="OUTCAR", lines=None):
+        """
+        Gets the ediel_sol for every ionic step from the OUTCAR file
+
+        Args:
+            filename (str): Filename of the OUTCAR file to parse
+            lines (list/None): lines read from the file
+
+        Returns:
+            numpy.ndarray: A 1xM array of the total energies in $eV$
+
+            where M is the number of time steps
+        """
+
+        def get_ediel_sol_from_line(line):
+            return float(_clean_line(line.strip()).split()[-1])
+
+        trigger_indices, lines = _get_trigger(
+            lines=lines,
+            filename=filename,
+            trigger="Solvation  Ediel_sol  = ",
+        )
+        return np.array([get_ediel_sol_from_line(lines[j]) for j in trigger_indices])
+
     @staticmethod
     def get_all_total_energies(filename="OUTCAR", lines=None):
         """
