From 4ae0b945cd55e096c844f0d8dee8dba3bab1f928 Mon Sep 17 00:00:00 2001
From: Jan Janssen <janssen@mpie.de>
Date: Tue, 4 Mar 2025 13:14:29 +0100
Subject: [PATCH] Use get_chemical_symbols() rather than get_parent_symbols()

---
 pyiron_vasp/vasp/output.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyiron_vasp/vasp/output.py b/pyiron_vasp/vasp/output.py
index d095b04..9a6700a 100644
--- a/pyiron_vasp/vasp/output.py
+++ b/pyiron_vasp/vasp/output.py
@@ -520,7 +520,7 @@ def get_final_structure_from_file(
         try:
             output_structure = read_atoms_funct(
                 filename=filename,
-                species_list=input_structure.get_parent_symbols(),
+                species_list=input_structure.get_chemical_symbols(),
             )
             input_structure.cell = output_structure.cell.copy()
             input_structure.positions[sorted_indices] = output_structure.positions
