From 9499b4d244ec6cde4c5a31f078c5ec0dcc6c4eec Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Fri, 28 Feb 2025 13:29:54 +0100
Subject: [PATCH] vasp output

---
 pyiron_vasp/vasp/output.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/pyiron_vasp/vasp/output.py b/pyiron_vasp/vasp/output.py
index dbf1020..d095b04 100644
--- a/pyiron_vasp/vasp/output.py
+++ b/pyiron_vasp/vasp/output.py
@@ -415,6 +415,7 @@ def parse_vasp_output(
     read_atoms_funct: callable = read_atoms,
     es_class=ElectronicStructure,
     bader_class=Bader,
+    output_parser_class=Output,
 ) -> dict:
     """
     Parse the VASP output in the working_directory and return it as hierachical dictionary.
@@ -427,7 +428,7 @@ def parse_vasp_output(
     Returns:
         dict: hierarchical output dictionary
     """
-    output_parser = Output()
+    output_parser = output_parser_class()
     if structure is None or len(structure) == 0:
         try:
             structure = get_final_structure_from_file(
